bcda/api/requests.go

```diff
@@ -242,10 +242,13 @@ func (h *Handler) JobStatus(w http.ResponseWriter, r *http.Request) {
 	job, jobKeys, err := h.Svc.GetJobAndKeys(r.Context(), uint(jobID))
 	if err != nil {
 		logger.Error(err)
-		// NOTE: This is a catch all and may not necessarily mean that the job was not found.
-		// So returning a StatusNotFound may be a misnomer
-		//In contrast to above, if the input COULD be valid, we return a not found header (404)
-		h.RespWriter.Exception(r.Context(), w, http.StatusNotFound, responseutils.DbErr, "")
+		if err == sql.ErrNoRows {
```

Comment:
Do we need to differentiate between an error from `GetJobById` and `GetJobKeys`? It seems like GetJobById should be StatusNotFound, and GetJobKeys should be an internal error since we'd expect job keys for successfully-completed jobs.

Reply:
Right before GetJobKeys it checks for the completed one, so that query won't end up executing if the job is completed. Since GetJobKeys uses QueryContext it _shouldn't_ return an error if there are no rows, since QueryContext returns an EOF if empty (vs QueryRowContext which throws the ErrNoRows), and for jobs that are Pending but not yet In Process, it's possible to have no job keys

Reply:
got it, thanks!